ipyleaflet/leaflet.py

```diff
@@ -1104,32 +1110,97 @@ def add_layer(self, layer):
     def remove_layer(self, rm_layer):
         """Remove a layer from the group.
 
+        .. deprecated :: 0.17.0
+           Use remove method instead.
+
         Parameters
         ----------
         layer: layer instance
             The layer to remove from the group.
         """
+        warnings.warn("remove_layer will be deprecated in future version, use remove instead", PendingDeprecationWarning)
         if rm_layer.model_id not in self._layer_ids:
             raise LayerException('layer not on in layergroup: %r' % rm_layer)
         self.layers = tuple([layer for layer in self.layers if layer.model_id != rm_layer.model_id])
 
     def substitute_layer(self, old, new):
         """Substitute a layer with another one in the group.
 
+        .. deprecated :: 0.17.0
+           Use substitute method instead.
+
         Parameters
         ----------
         old: layer instance
             The layer to remove from the group.
         new: layer instance
             The new layer to include in the group.
         """
+        warnings.warn("substitute_layer will be deprecated in future version, substitute instead", PendingDeprecationWarning)
+
         if isinstance(new, dict):
             new = basemap_to_tiles(new)
         if old.model_id not in self._layer_ids:
             raise LayerException('Could not substitute layer: layer not in layergroup.')
         self.layers = tuple([new if layer.model_id == old.model_id else layer for layer in self.layers])
 
     def clear_layers(self):
+        """Remove all layers from the group.
+
+        .. deprecated :: 0.17.0
+           Use clear method instead.
+
+        """
+
+        warnings.warn("clear_layers will be deprecated in future version, use clear instead", PendingDeprecationWarning)
+
+        self.layers = ()
+
+    def add(self, layer):
+        """Add a new layer to the group.
+
+        Parameters
+        ----------
+        layer: layer instance
+            The new layer to include in the group.
```

Comment:
The doc for `add`, `remove` etc doesn't say it can be used for controls as well.

```suggestion
    def add(self, layer_or_control):
        """Add a new layer or a control to the group.

        Parameters
        ----------
        layer_or_control: Layer or Control instance
            The new layer or control to include in the group.
```